packages/cli/src/commands/search/dump.ts

```diff
@@ -0,0 +1,169 @@
+import {Command, flags} from '@oclif/command';
+import {AuthenticatedClient} from '../../lib/platform/authenticatedClient';
+import {
+  Preconditions,
+  IsAuthenticated,
+} from '../../lib/decorators/preconditions/';
+import {Config} from '../../lib/config/config';
+import PlatformClient from '@coveord/platform-client';
+import {writeFile} from 'fs-extra';
+import {Parser} from 'json2csv';
+import {cli} from 'cli-ux';
+import {
+  buildAnalyticsFailureHook,
+  buildAnalyticsSuccessHook,
+} from '../../hooks/analytics/analytics';
+
+interface SearchResult {
+  raw: {rowid: string};
+}
+
+interface ProgressBar {
+  update: (up: number) => void;
+  stop: () => void;
+  start: (total: number, current: number) => void;
+  setTotal: (tot: number) => void;
+  increment: (inc: number) => void;
+}
+
+interface SearchResponse {
+  indexToken: string;
+  results: SearchResult[];
+  totalCount: number;
+}
+
+interface FetchParameters {
+  client: PlatformClient;
+  organizationId: string;
+  source: string;
+  additionalFilter: string;
+}
+
+export default class Dump extends Command {
+  static description =
+    'Dump the whole content of a particular source in a CSV format.';
+
+  static flags = {
+    source: flags.string({
+      char: 's',
+      description:
+        'The identifier of the source for which to extract all documents.',
+      helpValue: 'mySourceName',
+      required: true,
+    }),
+    destination: flags.string({
+      char: 'd',
+      description:
+        'The folder destination where the CSV file should be created',
+      default: '.',
+    }),
+    name: flags.string({
+      char: 'd',
+      description: 'The name of the CSV file that should be created',
+      default: 'indexdump',
+    }),
+    additionalFilter: flags.string({
+      char: 'f',
+      description:
+        'Additional search filter that should be applied while doing the extraction. See https://docs.coveo.com/en/1552 for more information',
+      default: '',
+    }),
+  };
+
+  @Preconditions(IsAuthenticated())
+  async run() {
+    const {flags} = this.parse(Dump);
+    const client = await new AuthenticatedClient().getClient();
+    const organizationId = (
+      await new Config(this.config.configDir, this.error).get()
+    ).organization;
+
+    const allResults = await this.fetchResults({
+      client,
+      organizationId,
+      source: flags.source,
+      additionalFilter: flags.additionalFilter,
+    });
+
+    if (allResults.length === 0) {
+      this.log(
+        'Found no results. Are you sure the source name and filter are valid ?'
+      );
+    } else {
+      const data = allResults.map((r) => r.raw);
+      const parser = new Parser({fields: Object.keys(allResults[0].raw)});
```

Comment:
ouf. The 2 above 2 lines might take a lot of memory if the index has a couple 100K documents.
Maybe a writeStream would be more appropriate here... or just dumping into the CSV after each call.

Reply:
Ive ran it on a source with 50k documents and it was fine. 
Took ~1min to query/download, but the file creation was pretty instant.

It took a while to open the CSV with excel though :D 

I could split on chunk, however.

Say, 10k documents maximum per file.

Reply:
correct. that's specially the case in commerce implementations with large spec fields...
or the infamous "all metadata fields".

Reply:
Chunk division is not a bad idea.
Also from a perf perspective, a writeStream is also a good idea, keeping an handle on the file while writting on it will be more efficient.
However you have to be absolutely sure to properly close it, even in case of failure. Otherwise it's a fd/socket that may stay open (although node does its best effort to close this kind of stuff in case of dirty closure, it's not perfect)

Reply:
Went with chunks, 10k results by default, possibility to exclude fields.